framework/Web/UI/TControl.php

```diff
@@ -346,6 +351,32 @@ public function getSourceTemplateControl()
 		return $this->getPage();
 	}
 
+	/**
+	 * Returns the module associated with the class path of the control.  This is for Composer
+	 * extensions adding their own Controls to access their associated Module.
+	 * @return null|Prado\Util\IPluginModule the module associated with this TControl
+	 * @since 4.2.0
+	 */
+	public function getPluginModule()
+	{
+		if ($this->_pluginmodule === false) {
+			$this->_pluginmodule = null;
+			foreach ($this->getApplication()->getModulesByType('Prado\\Util\\IPluginModule') as $id => $module) {
+				if (!$module) {
+					continue;
+				}
+				$reflect = new ReflectionClass(get_class($this));
+				$folder = $reflect->getFileName();
```

Comment:
Move these two lines out of the foreach loop to avoid executing them multiple times